nuxeo-connect-client/src/main/java/org/nuxeo/connect/registration/RegistrationHelper.java

```diff
@@ -68,73 +85,93 @@ protected static String getBaseUrl() {
+
+        // Set credentials provider
+        CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
         if (login != null) {
-            httpClient.getParams().setAuthenticationPreemptive(true);
             Credentials ba = new UsernamePasswordCredentials(login, password);
-            httpClient.getState().setCredentials(
-                    new AuthScope(null, -1, AuthScope.ANY_REALM), ba);
+            credentialsProvider.setCredentials(AuthScope.ANY, ba);
         }
+        context.setCredentialsProvider(credentialsProvider);
+
+        // Create AuthCache instance for preemptive authentication
+        AuthCache authCache = new BasicAuthCache();
+        // Generate BASIC scheme object and add it to the local auth cache
+        BasicScheme basicAuth = new BasicScheme();
+        try {
+            authCache.put(URIUtils.extractHost(new URI(url)), basicAuth);
+        } catch (URISyntaxException e) {
+            log.debug(e, e);
+        }
+        context.setAuthCache(authCache);
+
+        // Create request configuration
+        RequestConfig.Builder requestConfigBuilder = RequestConfig.custom().setConnectTimeout(10000);
+
         // Configure the http proxy if needed
-        ProxyHelper.configureProxyIfNeeded(httpClient, url);
-    }
+        ProxyHelper.configureProxyIfNeeded(requestConfigBuilder, credentialsProvider, url);
 
-    protected static HttpClient newHttpClient(String url, String login, String password) {
-        HttpClient httpClient = new HttpClient();
-        configureHttpClient(httpClient, url, login, password);
-        return httpClient;
+        context.setRequestConfig(requestConfigBuilder.build());
+        return context;
     }
 
-    public static List<ConnectProject> getAvailableProjectsForRegistration(
-            String login, String password) {
+    public static List<ConnectProject> getAvailableProjectsForRegistration(String login, String password) {
         String url = getBaseUrl() + GET_PROJECTS_SUFFIX;
-        HttpClient httpClient = newHttpClient(url, login, password);
-        HttpMethod method = new GetMethod(url);
         List<ConnectProject> result = new ArrayList<>();
-        try {
-            int rc = httpClient.executeMethod(method);
-            if (rc == HttpStatus.SC_OK) {
-                String json = method.getResponseBodyAsString();
-                JSONArray array = new JSONArray(json);
-                for (int i = 0; i < array.length(); i++) {
-                    JSONObject ob = (JSONObject) array.get(i);
-                    result.add(AbstractJSONSerializableData.loadFromJSON(
-                            ConnectProject.class, ob));
+        try (CloseableHttpClient httpClient = HttpClients.createDefault()) {
+            HttpGet method = new HttpGet(url);
+            try (CloseableHttpResponse httpResponse = httpClient.execute(method,
+                    getHttpClientContext(url, login, password))) {
```

Comment:
Move `method` declaration outside the `try` and collapse the two `try` into one.